src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java

```diff
@@ -948,6 +955,10 @@ public Double getTargetPos() {
   @Override
   public void setPosition(Double pos) {
     currentPos = targetPos = pos;
+    if (encoder != null) {
+      if (encoder instanceof TimeEncoder )
+      ((TimeEncoder)encoder).setPos(pos);
+    }
```

Comment:
sort of ugly - but probably "expected" behavior

Reply:
we can revisit this, but for now, i think we want to be pretty explicit about how certain features are enabled on different types of encoders.